docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,108 @@
+
+Currently scripts are written to be easily usable by GCE, but it should be relatively straightforward to port them to different providers or bare metal.
+
+## Common use cases and helper scripts
+
+Common workflow for Kubemark is:
+- starting a Kubemark cluster (on GCE)
+- running Density tests on Kubemark cluster
+- monitoring test execution and debugging problems
+-turning down Kubemark cluster
+
+In the text there will be comments helpful for anyone who’ll want to port Kubemark to different providers.
+
+### Starting a Kubemark cluster
+
+To start a Kubemark cluster on GCE you need to create an external cluster (it can be GCE or GKE cluster) by yourself and run `test/kubemark/start-kubemark.sh` script. It will do all the setup for you using the configuration stored in `cluster/kubemark/config-default.sh` - you can tweak it however you want, but note that some features may not implemented yet, as implementation of Hollow components/mocks will probably be lagging behind ‘real’ one. For performance tests interesting variables are `NUM_MINIONS` and `MASTER_SIZE`. After start-kubemark script is finished you’ll have a ready Kubemark cluster, script stores a kubeconfig file for talking to the Kubemark cluster in `test/kubemark/kubeconfig.loc`.
+
+Currently single HollowNode uses 50 millicores of CPU and ~60MB or memory, which taking into account default cluster addons and fluentD, allows running ~17.5 HollowNodes per core.
+
+#### Behind the scene details:
+
+Start-kubemark script does quite a lot of things:
+- Creates a master machine called hollow-cluster-master and PD for it (uses gcloud, should be easy to do outside of GCE)
+- Creates a firewall rule which opens port 443 on the master machine (uses gcloud, should be easy to do outside of GCE)
+- Builds a Docker image for HollowNode from the current repository and pushes it to the Docker repository (GCR for us, using scripts from `cluster/gce/util.sh` - it may get tricky outside of GCE)
+- Generates certificates and kubeconfig files, writes a kubeconfig locally to `test/kubemark/kubeconfig.loc` and creates a Secret which stores kubeconfig for HollowKubelet/HollowProxy use (used gcloud to transfer files to Master, should be easy to do outside of GCE).
+- Creates a ReplicationController for HollowNodes and starts them up. (will work exactly the same everywhere as long as MASTER_IP will be populated correctly, but you’ll need to update docker image address if you’re not using GCR and default image name)
+
+### Running Density tests on Kubemark cluster
+
+To run standard density ‘30 test on your Kubemark cluster created in the previous step you execute `test/kubemark/run-scalability-test.sh` script. It will configure ginkgo to use Kubemark cluster instead of something else and start a density test. This script should not need any changes to work on other cloud providers.
```

Comment:
I think that there is nothing special in density test (other than it doesn't require running these containers for correctness and not using volumes).
I think that what we should do is to change the script to take a regexp instead (by default we can use 30` Density) thanks to which we will be able to easily run other tests too.